### DISCUSSION TITLE
photopea.com: ads

### DISCUSSION THREAD

**Discussion by wickles** [uBlockOrigin/uAssets#19970] [General]
### Prerequisites

- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Use this [button](https://user-images.githubusercontent.com/585534/146582579-c32707a0-36a1-4cc5-ad50-83172c9f67a8.png) to find out.)
- [X] I forced an update of my filter lists. (Click the ["Purge all caches"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#purge-all-caches) button while holding the 'Shift' key, then click the ["Update now"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#update-now) button.)
- [X] I did not remove any of the [default filter lists](https://user-images.githubusercontent.com/124740436/235392297-c4fc6290-73a6-4a70-96a9-3f6a7683f21d.png), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin (uBO) along with other content blocker extensions.
- [X] I have verified that the web browser's built-in blocker or [DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking) (standalone or through a VPN) is not causing the issue.
- [ ] I did not answer truthfully to **ALL** the above checkpoints.

### URL address of the web page

`https://www.photopea.com/`

### Category

ads

### Description

photopea.com inserts a sidebar for ads. Ublock removes the ads but not the sidebar. I am unable to remove it with the element picker/zapper. 

### Other extensions used

none

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![Screenshot 2023-09-30 121326](https://github.com/uBlockOrigin/uAssets/assets/4229542/5fcb1815-8192-47d2-9eeb-f164c090efd9)

</details>

### Configuration

<details>

```yaml
uBlock Origin: 1.52.2
Firefox: 118
filterset (summary):
 network: 142799
 cosmetic: 248416
 scriptlet: 45426
 html: 1242
listset (total-discarded, last-updated):
 added:
  https://gitlab.com/magnolia1234/bypass-paywalls-clean-filters/-/raw/main/bpc-paywall-filter.txt: 948-0, 3m
  adguard-generic: 77828-5722, 3m
  block-lan: 54-0, 3m
  adguard-social: 21557-867, 3m
  adguard-cookies: 27725-67, 3m
  adguard-popup-overlays: 24502-1636, 3m
  adguard-mobile-app-banners: 4644-7, 3m
  adguard-other-annoyances: 13606-611, 3m
  adguard-widgets: 2311-2, 3m
  fanboy-thirdparty_social: 68-0, 3m
  easylist-annoyances: 3985-119, 3m
  easylist-chat: 139-11, 3m
  [9 lists not shown]: [too many]
 default:
  user-filters: 0-0, never
  ublock-filters: 35769-48, 3m
  ublock-badware: 7886-0, 3m
  ublock-privacy: 586-0, 3m
  ublock-unbreak: 2172-1, 3m
  ublock-quick-fixes: 241-91, 3m
  easylist: 71296-714, 3m
  easyprivacy: 33196-878, 3m
  urlhaus-1: 3841-0, 3m
  plowe-0: 3720-1, 3m
filterset (user): [empty]
userSettings:
 cloudStorageEnabled: true
hiddenSettings: [none]
supportStats:
 allReadyAfter: 645 ms (selfie)
 maxAssetCacheWait: 228 ms
popupPanel:
 blocked: 1
 network:
  googletagmanager.com: 1
 extended:
  ##a[href*="photopea.com"][href*="ads"]
  ##[href*=".ivank.net"]
  ##div[style^="padding-top: 10px; overflow: hidden; padding-left:…
  ##+js(set-constant, console.clear, noopFunc)
  ##+js(no-fetch-if, /pagead2\.googlesyndication\.com|cmp\.unicons…
```
</details>

**Comment by mapx-** [comment]
maybe 

`photopea.com##.body, .storage:style(width: 1800px !important)`

**Comment by wickles** [comment]
> maybe
> 
> `photopea.com##.body, .storage:style(width: 1800px !important)`

Messes with the site functionality -- editor sidebar goes off screen.

**Comment by mapx-** [comment]
try modifying the width value to catch your settings

**Comment by wickles** [comment]
Looks OK at first glance with `1500px` but it messes with the sidebar in other ways too.

<details>
<summary>Screenshot: without filter</summary>

![Screenshot 2023-10-02 085030](https://github.com/uBlockOrigin/uAssets/assets/4229542/8fada5c5-c25d-4d47-8083-c0276afae336)

</details>

<details>
<summary>Screenshot: with filter</summary>

![Screenshot 2023-10-02 085047](https://github.com/uBlockOrigin/uAssets/assets/4229542/3ab808e3-6648-4e68-a491-b74c2df8ab78)

</details>

<details>
<summary>Screenshot: with filter + expando</summary>

![Screenshot 2023-10-02 085053](https://github.com/uBlockOrigin/uAssets/assets/4229542/9334c813-3bd5-4786-a001-814b8e957aad)

</details>

**Comment by krystian3w** [comment]
Maybe needed is again `rpnt` scriptlet operating on the "tiny" border of Opera, Mozilla and Microsoft store regulations.

Google like raw JavaScript as filter.


Older idea by MasterKia and his team:

```adb
!#if !env_mobile
photopea.com##+js(rpnt, script, /^.*?(function gtag).*$/, window.innerWidth = document.documentElement.clientWidth + 320;)
!#endif
```

Still can works bad delete whole `/^.*?(function gtag).*$/` or default "loop" in scriptlet if exist (I found once one blog with rapair DOM Tree after user damaged something).

**Comment by MasterKia** [comment]
https://github.com/jared-schwalbe/remove-ads-from-photopea/issues/5

cc @jared-schwalbe
